v1.8.7 下 Wireguard 使用 IPv6 作为 endpoint 会导致 panic

```jsonc
{
  "inbounds": [
    {
      "port": 8080,
      "protocol": "dokodemo-door",
      "settings": {
        "address": "1.1.1.1",
        "port": 80
      }
    }
  ],
  "outbounds": [
    {
      "tag": "WARP",
      "protocol": "wireguard",
      "settings": {
        "secretKey": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=",
        "peers": [
          {
            "publicKey": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=",
            "allowedIPs": [
              "0.0.0.0/0",
              "::/0"
            ],
            "endpoint": "[2000:1234::1]:2408"
          }
        ]
      }
    }
  ]
}
```
报错内容：
```console
Xray 1.8.7 (Xray, Penetrates Everything.) 3f0bc13 (go1.21.5 linux/amd64)
A unified platform for anti-censorship.
2024/01/25 21:39:02 [Info] infra/conf/serial: Reading config: ./config.json
2024/01/25 21:39:02 [Warning] core: Xray 1.8.7 started
2024/01/25 21:39:04 [Error] IPC error -22: failed to set endpoint [2000:1234: address [2000:1234: missing ']' in address
panic: close of closed channel

goroutine 68 [running]:
github.com/xtls/xray-core/proxy/wireguard/gvisortun.(*netTun).Close(0xc00033d710)
        github.com/xtls/xray-core/proxy/wireguard/gvisortun/tun.go:180 +0x38
github.com/xtls/xray-core/proxy/wireguard.(*tunnel).Close(0xc00033c4b0)
        github.com/xtls/xray-core/proxy/wireguard/tun.go:89 +0xc9
github.com/xtls/xray-core/proxy/wireguard.(*gvisorNet).Close(0xc0001c7960?)
        github.com/xtls/xray-core/proxy/wireguard/tun.go:127 +0x15
github.com/xtls/xray-core/proxy/wireguard.(*Handler).makeVirtualTun(0xc0001c7960, 0xc00035a9b0)
        github.com/xtls/xray-core/proxy/wireguard/client.go:251 +0xd8
github.com/xtls/xray-core/proxy/wireguard.(*Handler).processWireGuard(0xc0001c7960, {0x14ef620?, 0xc000343880})
        github.com/xtls/xray-core/proxy/wireguard/client.go:120 +0x2e5
github.com/xtls/xray-core/proxy/wireguard.(*Handler).Process(0xc0001c7960, {0x14f2068, 0xc00033c390}, 0xc00009a200, {0x14ef620, 0xc000343880})
        github.com/xtls/xray-core/proxy/wireguard/client.go:140 +0x128
github.com/xtls/xray-core/app/proxyman/outbound.(*Handler).Dispatch(0xc000343880, {0x14f2068, 0xc00033c390}, 0xc00009a200)
        github.com/xtls/xray-core/app/proxyman/outbound/handler.go:206 +0x3f8
github.com/xtls/xray-core/app/dispatcher.(*DefaultDispatcher).routedDispatch(0xc00032d560, {0x14f2068, 0xc00033c390}, 0xc00009a200, {{0x14f21f0, 0xc000327268}, 0x50, 0x2})
        github.com/xtls/xray-core/app/dispatcher/default.go:443 +0xbc6
created by github.com/xtls/xray-core/app/dispatcher.(*DefaultDispatcher).Dispatch in goroutine 67
        github.com/xtls/xray-core/app/dispatcher/default.go:236 +0x4f6

```

--- a/proxy/wireguard/client.go
+++ b/proxy/wireguard/client.go
@@ -254,7 +254,6 @@ func (h *Handler) makeVirtualTun(bind *netBindClient) (Tunnel, error) {
 	return t, nil
 }
 
-
 // serialize the config into an IPC request
 func (h *Handler) createIPCRequest(bind *netBindClient, conf *DeviceConfig) string {
 	var request strings.Builder
@@ -275,8 +274,11 @@ func (h *Handler) createIPCRequest(bind *netBindClient, conf *DeviceConfig) stri
 			request.WriteString(fmt.Sprintf("preshared_key=%s\n", peer.PreSharedKey))
 		}
 
-		split := strings.Split(peer.Endpoint, ":")
-		addr := net.ParseAddress(split[0])
+		address, port, err := net.SplitHostPort(peer.Endpoint)
+		if err != nil {
+			newError("failed to split endpoint ", peer.Endpoint, " into address and port").AtError().WriteToLog()
+		}
+		addr := net.ParseAddress(address)
 		if addr.Family().IsDomain() {
 			dialerIp := bind.dialer.DestIpAddress()
 			if dialerIp != nil {
@@ -306,7 +308,7 @@ func (h *Handler) createIPCRequest(bind *netBindClient, conf *DeviceConfig) stri
 		}
 
 		if peer.Endpoint != "" {
-			request.WriteString(fmt.Sprintf("endpoint=%s:%s\n", addr, split[1]))
+			request.WriteString(fmt.Sprintf("endpoint=%s:%s\n", addr, port))
 		}
 
 		for _, ip := range peer.AllowedIps {

